tempesta_fw/http.c

```diff
@@ -3930,7 +3931,9 @@ tfw_http_msg_process(void *conn, TfwFsmData *data)
 		if (likely(r == T_OK || r == T_POSTPONE)) {
 			data->skb->next = data->skb->prev = NULL;
 			data->trail = !next ? trail : 0;
-			r = __tfw_http_msg_process(conn, data);
+			r = TFW_CONN_TLS((TfwConn *)conn) && tfw_http2_context(conn)
+				? tfw_http2_frame_process(conn, data)
+				: tfw_http_msg_process_generic(conn, data);
```

Comment:
>Check H2 state in tfw_http_req_process()?
Yeah - this must be a frame type and, maybe, a stream state (since we have to track request's start/end, and this exactly matches with stream's states, as mentioned in penultimate paragraph in https://tools.ietf.org/html/rfc7540#section-8.1).
But currently I'm not very optimistic about the parser division into separate parts: indeed, we have rather effective implementation with single-jump cost to enter any state; and with divided parser we will get costs of additional function calls (if not inlined) and necessity for additional branching to call separate parser parts (with associated costs). But in general, I am still thinking about this point.